tests/framework/utils.py

```diff
@@ -760,16 +762,38 @@ def start_screen_process(screen_log, session_name, binary_path, binary_params):
         delay=1,
     ).group(1)
 
-    binary_clone_pid = int(
-        open("/proc/{0}/task/{0}/children".format(screen_pid), encoding="utf-8")
-        .read()
-        .strip()
-    )
-
     # Configure screen to flush stdout to file.
     flush_cmd = 'screen -S {session} -X colon "logfile flush 0^M"'
     run_cmd(flush_cmd.format(session=session_name))
 
+    # Make sure the screen process launched successfully
+    # As the parent process for the binary.
+    try:
+        retry_call(
+            confirm_process_running,
```

Comment:
Could we use psutil here?

```python
screen_ps = psutil.Process(screen_pid)
screen_ps.is_running()
```

Reply:
I'll give this a go, although there is no retry here in case the process hasn't come up yet. But I think this implementation will be much cleaner.